src/libs/ajax/data-table-providers/WdsDataTableProvider.test.ts

```diff
@@ -700,20 +714,64 @@ describe('transformMetadata', () => {
   })
 })
 
-describe('getWdsUrl', () => {
+
+describe('resolveWdsUrl', () => {
   it('properly extracts the proxy Url from the leo response', () => {
-    expect(getWdsUrl(testProxyUrlResponse)).toBe(testProxyUrl)
+    expect(resolveWdsUrl(testProxyUrlResponse, uuid)).toBe(testProxyUrl)
   })
-  it('locate the Url when the app name is different and is running', () => {
+  it('return an empty string if WDS is still PROVISIONING', () => {
+    jest.mock('./WdsDataTableProvider', () => ({ createLeoAppWithErrorHandling: () => {} }
+    ))
     const testProxyUrlResponseWithDifferentAppName: Array<Object> = [
-      { appType: 'CROMWELL', appName: 'something-else', status: 'RUNNING', proxyUrls: { wds: testProxyUrl } }
+      { appType: 'CROMWELL', appName: 'something-else', status: 'PROVISIONING', proxyUrls: { wds: testProxyUrl } }
     ]
-    expect(getWdsUrl(testProxyUrlResponseWithDifferentAppName)).toBe(testProxyUrl)
+    expect(resolveWdsUrl(testProxyUrlResponseWithDifferentAppName, uuid)).toBe('')
   })
   it('return empty string when app not found', () => {
     const testProxyUrlResponseWithDifferentAppName: Array<Object> = [
       { appType: 'A_DIFFERENT_APP', appName: 'something-else', status: 'RUNNING', proxyUrls: { wds: testProxyUrl } }
     ]
-    expect(getWdsUrl(testProxyUrlResponseWithDifferentAppName)).toBe('')
+    expect(resolveWdsUrl(testProxyUrlResponseWithDifferentAppName, uuid)).toBe('')
+  })
+  it('return an empty RUNNING app if more than one exists', () => {
```

Comment:
nit: this test name seems wrong, it should be something like `'return empty string if no CROMWELL app exists but other apps are present'`

Reply:
Updated!